extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/TracerRuntimeConfig.java

```diff
@@ -18,23 +19,36 @@
     @ConfigItem
     Optional<List<String>> resourceAttributes;
 
-    /**
-     * The sampler to use for tracing
-     * <p>
-     * Valid values are {@code always_off, always_on, parentbased_always_off,
-     * parentbased_always_on, parentbased_traceidratio, traceidratio}.
-     * <p>
-     * Defaults to {@code parentbased_always_on}.
-     */
-    @ConfigItem(defaultValue = "parentbased_always_on")
-    public String sampler;
+    /** Config for sampler */
+    public SamplerConfig sampler;
 
-    /**
-     * The sampler ratio to use for tracing.
-     * <p>
-     * Only supported by the {@code parentbased_traceidratio, traceidratio}
-     * samplers.
-     */
-    @ConfigItem(name = "sampler.ratio")
-    public Optional<Double> samplerRatio;
+    @ConfigGroup
+    public static class SamplerConfig {
+        /**
+         * The sampler to use for tracing
+         * <p>
+         * Valid values are {@code off, on, ratio}.
+         * <p>
+         * Defaults to {@code on}.
+         */
+        @ConfigItem(name = ConfigItem.PARENT, defaultValue = "on")
+        public String samplerName;
+
+        /**
+         * The sampler ratio to use for tracing.z
```

Comment:
Is ".z" a carryover from a previous comment?

Reply:
Dunno 🤦 , maybe just me with my crazy hands hahah